events/EventQueue.cpp

```diff
@@ -74,9 +75,9 @@ void EventQueue::background(Callback<void(int)> update)
 void EventQueue::chain(EventQueue *target)
 {
     if (target) {
-        equeue_chain(&_equeue, &target->_equeue);
+        MBED_ASSERT(equeue_chain(&_equeue, &target->_equeue) == 0);
```

Comment:
No, this eliminates the equeue_chain call in a release build!

Not sure about the assert in principle - out of memory is a runtime error, not a programming error, so still needs to be checked in release builds. `new` leads to `MBED_ERROR` not `MBED_ASSERT` for that reason.

And also, it leaves it inconsistent with the rest of the API, which does return error codes (whether people check them is another matter).

So my two acceptable options are:
* "throw" errors with `MBED_ERROR`
* return errors

Reply:
Fixed - Returning error.